### PR TITLE
World: Add a config file for setting the API URL

### PR DESCRIPTION
**Pull request by tombettany**
Allow the API URL to be set via a configuration file to allow system
testing to be done.

### CODE REVIEW

**Comment by skarbat** [comment]
FYI: Built on the RPi, all `test-library` tests pass.

File: include/mercury/kw/APIConfig.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    /**
+     * \brief Get a value from the configuration
+     *
+     * \param key    They value's key
+     *
+     * \returns the key's value
+     */
+    template<typename T>
+    T get_value(const std::string& key) const {
+        for (const auto&config : this->configs) {
```
Comment (radujipa) [inline include/mercury/kw/APIConfig.h]:
```suggestion
        for (const auto& config : this->configs) {
```

File: include/mercury/kw/APIConfig.h [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    /**
+     * \brief Get a value from the configuration
+     *
+     * \param key    They value's key
+     *
+     * \returns the key's value
+     */
+    template<typename T>
+    T get_value(const std::string& key) const {
```
Comment (radujipa) [inline include/mercury/kw/APIConfig.h]: Does the implementation of this function have to be in the header?

File: include/mercury/kw/APIConfig.h [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+namespace Mercury {
+namespace KanoWorld {
+
+
+/**
+ * \class APIConfig
+ * \brief Provides configurations for the API configurations
+ */
+class APIConfig {
```
Comment (radujipa) [inline include/mercury/kw/APIConfig.h]: `IAPIConfig`?

File: include/mercury/kw/APIConfig.h (hunk 2 above)
Comment (radujipa) [reply include/mercury/kw/APIConfig.h]: Also, I could see a `has(const std::string& key)` method being useful here.

File: src/kw/APIConfig.cpp [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+using std::vector;
+
+
+APIConfig::APIConfig(vector<string> config_files) :
+        API_URL_KEY("API_URL"),
+        // FIXME: Find a cross-platform mechanism for this
+        SYSTEM_PATH("/etc/mercury-api.yaml") {
+#if CMAKE_BUILD_TYPE == Debug
+    config_files.push_back(
+        string(CMAKE_PROJ_BASE_DIR) + "conf/dev.yaml");
```
Comment (radujipa) [inline src/kw/APIConfig.cpp]: Maybe make this yaml specific to the module?
```suggestion
        string(CMAKE_PROJ_BASE_DIR) + "conf/kw_dev.yaml");
```

File: include/mercury/kw/APIConfig.h (hunk 2 above)
Comment (tombettany) [reply include/mercury/kw/APIConfig.h]: Added

File: include/mercury/kw/APIConfig.h (hunk 3 above)
Comment (tombettany) [reply include/mercury/kw/APIConfig.h]: Definitely want to do this long-term but maybe let's wait until we figure out how we want configs to work across the board here before splitting out what the interface is.

**Comment by tombettany** [comment]
Will merge as-is for now. When we've decided what config files look like we can address the config interface.

**Pull request merged**